content/docs/concepts/remote_write_spec_2_0.md

```diff
@@ -0,0 +1,394 @@
+---
+title: Prometheus Remote-Write Specification 2.0
+sort_rank: 4
+---
+
+# Prometheus Remote-Write Specification
+
+* Version: 2.0
+* Status: Proposed
+* Date: May 2024
+
+The remote write specification, in general, is intended to document the standard for how Prometheus and Prometheus remote-write-compatible agents send data to a Prometheus or Prometheus remote-write compatible receivers.
+
+This document is intended to define a second version of the [Prometheus Remote Write](./remote_write_spec.md) API with minor changes to protocol and semantics. This second version also adds a new wire format with new features enabling more use cases and wider adoption on top of performance and cost savings. Finally, this spec outlines how to implement backward compatible senders and receivers (even under a single endpoint) using existing basic content negotiation request headers. More advanced, automatic content negotiation mechanisms might come in future versions, if needed. For the rationales behind the 2.0 specification, see [the formal proposal](https://github.com/prometheus/proposals/pull/35).
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Introduction
+
+### Background
+
+The remote write protocol is designed to make it possible to reliably propagate samples in real-time from a sender to a receiver, without loss.
+
+The remote write protocol is designed to make stateless implementations of the server possible; as such there are little-to-no inter-message references.  As such the protocol is not considered "streaming." To achieve a streaming effect multiple messages should be sent over the same connection using e.g. HTTP/1.1 or HTTP/2. "Fancy" technologies such as gRPC were considered, but at the time were not widely adopted, and it was challenging to expose gRPC services to the internet behind load balancers such as an AWS EC2 ELB.
+
+The remote write protocol contains opportunities for batching, e.g. sending multiple samples for different series in a single request. It is not expected that multiple samples for the same series will be commonly sent in the same request, although there is support for this in the protocol.
+
+<!---
+TODO(bwplotka): Challenge last sentence (e.g. OTel coll usage)?
```

Comment:
Yeah, I think removing the current note but keeping some kind of note here would be preferable but I'm not sure about how to word it.

```
Note that while the remote write protocol technically allows for an applications to write metrics directly to Prometheus compatible recievers themselves remote write is not intended to be used this way. Prometheus compatible backends and some PromQL functions expect samples to be written to the storage at consistent intervals rather than on every event, like log lines, and as such writing metrics directly from your application could cause incompatibility problems.
```

I'm not sure if doing that :point_up: could actually cause any issues. I suppose it could also be easier to get out of order/duplicate issues sending metrics in this way.

Reply:
Yea, I think we are then not really sure what we try to share here. The fact that scraping is be preferred for Prometheus (which I agree) should be likely advertised in Prometheus, not in spec 🤔 

The only reason, I think, to have such statement, is to give intend that we won't add more spec features for pure push metric from monitored application use case,  but I can't imagine any example feature ATM. Should we remove this piece and add it back once we have a clear example of feature this spec will not implement?

> I suppose it could also be easier to get out of order/duplicate issues sending metrics in this way.

Hm, we don't specify anything about out of order - it's up to receiver. So perhaps not specific enough reason to add such intention paragraph?

Proposing removing for now 🤗